extensions/binding/org.eclipse.smarthome.binding.mqtt/src/main/java/org/eclipse/smarthome/binding/mqtt/internal/MqttBrokerHandlerFactory.java

```diff
@@ -46,14 +51,19 @@
  * @author David Graeff - Initial contribution
  */
 @NonNullByDefault
-@Component(immediate = true, service = { ThingHandlerFactory.class,
-        MQTTTopicDiscoveryService.class }, configurationPid = "MqttBrokerHandlerFactory")
-public class MqttBrokerHandlerFactory extends BaseThingHandlerFactory implements MQTTTopicDiscoveryService {
+@Component(immediate = true, service = { ThingHandlerFactory.class, MQTTTopicDiscoveryService.class,
+        MQTTBrokerConnectionDiscoveryService.class }, configurationPid = "MqttBrokerHandlerFactory")
+public class MqttBrokerHandlerFactory extends BaseThingHandlerFactory
+        implements MQTTTopicDiscoveryService, MQTTBrokerConnectionDiscoveryService {
```

Comment:
This is imho all VERY misleading. Neither `MQTTTopicDiscoveryService` nor `MQTTBrokerConnectionDiscoveryService` are ESH `DiscoveryService`s - I'd therefore suggest to use a different naming than "DiscoveryService" and "DiscoveryParticipant" - why not e.g. just have `MQTTBrokerConnectionListener`s?
It also feels wrong that the handler factory implements those interfaces at all - from what I see, this can be completely removed.

Reply:
Only the mqtt binding factory knows about all broker connections. The factory would be involved in any architecture (except if I iterate over the Thing registry, like I did before) or do I miss something?

The `MQTTTopicDiscoveryService` and `MQTTBrokerConnectionDiscoveryService` can be renamed, of course. We struggled last time to find a proper name.